lib/src/dependency_validator.dart

```diff
@@ -28,31 +28,28 @@ import 'pubspec_config.dart';
 import 'utils.dart';
 
 /// Check for missing, under-promoted, over-promoted, and unused dependencies.
-Future<void> run() async {
-  if (!File('pubspec.yaml').existsSync()) {
+Future<void> run({String root = "."}) async {
+  if (!File('$root/pubspec.yaml').existsSync()) {
     logger.shout(red.wrap('pubspec.yaml not found'));
     exit(1);
   }
-  if (!File('.dart_tool/package_config.json').existsSync()) {
-    logger.shout(red.wrap(
-        'No .dart_tool/package_config.json file found, please run "pub get" first.'));
-    exit(1);
-  }
 
   DepValidatorConfig config;
-  final configFile = File('dart_dependency_validator.yaml');
+  final configFile = File('$root/dart_dependency_validator.yaml');
   if (configFile.existsSync()) {
     config = DepValidatorConfig.fromYaml(configFile.readAsStringSync());
   } else {
     final pubspecConfig = PubspecDepValidatorConfig.fromYaml(
-        File('pubspec.yaml').readAsStringSync());
+      File('$root/pubspec.yaml').readAsStringSync(),
+    );
     if (pubspecConfig.isNotEmpty) {
       logger.warning(yellow.wrap(
           'Configuring dependency_validator in pubspec.yaml is deprecated.\n'
           'Use dart_dependency_validator.yaml instead.'));
     }
     config = pubspecConfig.dependencyValidator;
   }
+
   final excludes = config.exclude
```

Comment:
Realized this a bit after the fact, but since we're re-using the same config values as the root, these `exclude` paths will be re-evaluated on subpackages

Meaning, if we exclude:

```yaml
exclude:
  - lib/src/some_path/**
```

This would exclude `lib/src/some_path` in the root, and also exclude that same path in each subpackage

I think a better implementation for this is to treat each path as if it came from the root, which should be pretty easy to achieve using [`relative`](https://pub.dev/documentation/path/latest/path/relative.html) in conjunction with the `root` parameter